src/main/java/org/opensearch/flowframework/transport/WorkflowRequest.java

```diff
@@ -43,19 +43,36 @@ public class WorkflowRequest extends ActionRequest {
      * @param template the use case template which describes the workflow
      */
     public WorkflowRequest(@Nullable String workflowId, @Nullable Template template) {
-        this(workflowId, template, false);
+        this(workflowId, template, false, false);
     }
 
+    /**
+     * The all parameter on the get request
+     */
+    private boolean all;
+
     /**
      * Instantiates a new WorkflowRequest
      * @param workflowId the documentId of the workflow
      * @param template the use case template which describes the workflow
      * @param dryRun flag to indicate if validation is necessary
+     * @param all whether the get request is looking for all fields in status
      */
-    public WorkflowRequest(@Nullable String workflowId, @Nullable Template template, boolean dryRun) {
+    public WorkflowRequest(@Nullable String workflowId, @Nullable Template template, boolean dryRun, boolean all) {
```

Comment:
This `WorkflowRequest` class was meant primarily for creating/provisioning. I think it may be beneficial to create a separate request class for use in the status rest action rather than adding more fields to this. We won't be using this `all` field for creating/provisioning a workflow

Reply:
Created a GetWorkflowRequest class